### PR TITLE
Teamq/4502  Moving dates in the monthly/weekly calendar view

### PR DESCRIPTION
**Pull request by alfonsoLopez1992**
- The creator should be able to move the date in the monthly/weekly view. 
- Nobody else should be able to move the date.
- When the creator moves it he/she still has to confirm the edit by clicking on okay on the pop-up.

### CODE REVIEW

**Comment by EofChris** [comment]
Hi @alfonsoLopez1992,

can you please update the PR, because it also contains the changes from other tickets.

**Comment by alfonsoLopez1992** [comment]
Hi @EofChris , thank you for comments, please review tha changes, the comments are resolved .

**Review by EofChris — CHANGES_REQUESTED** [review]
Please use a SecurityVoter instead of checking rights directly in the controller. E.g. take a look at Security/Authorization/Voter/ItemVoter.php.
Either add to ItemVoter or create a new DateVoter if applicable. In the controller you can then make use of $this->isGranted(...).

**Comment by alfonsoLopez1992** [comment]
Hi @EofChris , I create new method in the class cs_dates_item.php, this method is similar at isPublic.

**Comment by alfonsoLopez1992** [comment]
Hi @EofChris , I create new method in the class cs_dates_item.php, this method is similar at isPublic.

**Comment by EofChris** [comment]
HI @alfonsoLopez1992 , maybe I was a little bit to loose on it. I don't want to have the access decision code neither in the DatesController nor the old leagacy cs_dates_item.

Please create a new symfony voter in `App\Security\Authorization\Voter\DatesVoter` (you can take the CalendarsVoter or one of the others as reference). Move your check into this class and update the DatesController to something like `'editable' => $this->isGranted(DatesVoter::EDIT, $this->legacyEnvironment->getCurrentUserItem()),`

**Review by EofChris — CHANGES_REQUESTED** [review]
HI @alfonsoLopez1992 , maybe I was a little bit to loose on it. I don't want to have the access decision code neither in the DatesController nor the old leagacy cs_dates_item.

Please create a new symfony voter in `App\Security\Authorization\Voter\DatesVoter` (you can take the CalendarsVoter or one of the others as reference). Move your check into this class and update the DatesController to something like `'editable' => $this->isGranted(DatesVoter::EDIT, $this->legacyEnvironment->getCurrentUserItem()),`

**Comment by alfonsoLopez1992** [comment]
Hi @EofChris , please check the changes for comments. Thak you, Please check the changes.

File: src/Security/Authorization/Voter/DateVoter.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+use App\Services\LegacyEnvironment;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+
+class DateVoter extends Voter
+{
+    const EDIT = 'edit';
+
+    private $legacyEnvironment;
```
Comment (EofChris) [inline src/Security/Authorization/Voter/DateVoter.php:12]:
```suggestion
    private LegacyEnvironment $legacyEnvironment;
```
Please add the type

File: config/services.yaml [hunk 2]
```diff
@@ -333,7 +333,12 @@ services:
         public: false
         tags:
             - { name: security.voter }
-
+    commsy.security.date_voter:
+        class: App\Security\Authorization\Voter\DateVoter
+        arguments: [ '@commsy_legacy.environment' ]
+        public: true
+        tags:
+            - { name: security.voter }
```
Comment (EofChris) [inline config/services.yaml]: Please remove the service definition, I don't think it is necessary.

**Review by EofChris — CHANGES_REQUESTED** [review]
Just two small changes, thanks

File: config/services.yaml (hunk 2 above)
Comment (alfonsoLopez1992) [reply config/services.yaml]: @EofChris please, check the changes.Comments resolved.

**Pull request merged**